src/rules/no-extraneous-dependencies.js

```diff
@@ -126,6 +126,36 @@ function getModuleRealName(resolved) {
   return getFilePackageName(resolved);
 }
 
+function checkDependencyDeclaration(deps, packageName) {
+  // in case of sub package.json inside a module
+  // check the dependencies on all hierarchy
+  const packageHierarchy = [];
+  const packageNameParts = packageName.split('/');
```

Comment:
on windows, this might need `.split(path.sep)`?

Reply:
PackageName variable is @types/react or lodash/get

It is always / independently of the platform